### ISSUE TITLE
"Invalid `# ruff: noqa`" warnings should be an error

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#8157]
```py
# ruff: noqa: asdfas
```
```
>ruff asdf.py
warning: Invalid `# ruff: noqa` directive at asdf.py:1: expected a comma-separated list of codes (e.g., `# noqa: F401, F841`).
```

currently there seems to be no way to make ruff fail on invalid noqa comments. i think this should be a normal rule that causes ruff to fail, not a warning.

**Comment by zanieb** [comment]
It does seems reasonable to include a rule code for this, but given that we cannot parse the noqa directive it could be troublesome as you won't be able to ignore problems.

Separately from whether or not this is something we want, I think it'd be a little tricky to implement.

**Comment by andy-maier** [comment]
I have this warning too, but in my case it was caused by the use of `noga: 501` which apparently works for flake8 but causes the warning for ruff. We addressed that by changing that to `noga: E501`.

However, since flake8 supports the notation `noga: 501`, it would probably be good if ruff did that, too.

Apart from that, I tend to agree that invalid noqa statements should be errors.

**Comment by DetachHead** [comment]
related: #15682

i think when i raised this only `# ruff: noqa:` comments had this issue, but now the same thing happens with `# noqa:` comments.

i think this warning can probably just be removed once those issues with `RUF100` are fixed

**Comment by KotlinIsland** [comment]
why don't these warnings show in the lsp? i don't think anyone would ever even notice that they exist otherwise

**Comment by KotlinIsland** [comment]
if a warning is emitted, will it still cause the ci to fail? I can't really see any use-case for it otherwise

**Comment by MichaReiser** [comment]
No, CI won't fail which is the same as for other warnings emitted by Ruff, e.g. warnings about incorrect or conflicting configurations, failures to read the cache, etc. This behavior is intentional (in the absence of an explicit warning severity and a `--error-on-warning` flag).

**Comment by MichaReiser** [comment]
One thing we could try is to see if we can setup a custom tracing subscriber that records if it has seen any warning log message, and, if so, return with an error exit code if `error-on-warning` is enabled in the configuration.
